sections/kubecontext.zsh

```diff
@@ -31,13 +31,28 @@ spaceship_kubecontext() {
   local kube_context=$(kubectl config current-context 2>/dev/null)
   [[ -z $kube_context ]] && return
 
+  local section_color
+
+  # Apply custom color to section if context name is defined in of SPACESHIP_KUBECONTEXT_COLORGROUPS KV map.
+  # See Options.md for usage example.
+  if [[ ${#SPACESHIP_KUBECONTEXT_COLORGROUPS[@]} -gt 0 ]]; then
+    local pattern color
+    for pattern color in ${(kv)SPACESHIP_KUBECONTEXT_COLORGROUPS}; do
+      if [[ "$kube_context" =~ "$pattern" ]]; then
+        section_color=$color
+        break
+      fi
+    done
+  fi
+  [[ -z "$section_color" ]] && section_color=$SPACESHIP_KUBECONTEXT_COLOR
```

Comment:
This is becoming very awesome in my opinion, nice job 👍 I suggest we move this entire block after namespace detection, so that users can additionally build rules for namespaces. For example, a rule might be `[kube-system]=red`.

Reply:
That's a good idea.

I am thinking though whether configuring it is straightforward enough, because it means that the pattern would need to depend on how the namespace is concatenated to the context name (i.e. `%s (%s)`) and whether it's even displayed or not (be it due to `SPACESHIP_KUBECONTEXT_NAMESPACE_SHOW` option or if it's "default").

Reply:
I think it's fine for now, my proposal would be the same as earlier, let's keep implementation simple so that it's just possible to define simple rules like `-prod-` and `kube-system`, as soon as someone needs more sophisticated stuff that isn't currently possible, we will implement it — your #575 is a perfect example how we can iterate. I simply want to avoid wasting your and my time and complicating codebase without a practical reason 😉

Reply:
So be it! :)

Reply:
Remember to add `[kube-system]=red` to the examples in `Options.md` too 😉

Reply:
Uhh so the order of the array is not determined by the way it was set, apparently.

```sh
[18:32:55] spaceship-prompt <k:dev11-system> $ typeset -A SPACESHIP_KUBECONTEXT_COLORGROUPS=(
  [dev-01]=green # Green if context or namespace have "dev-01" in them
  ['\(kube-system\)$']=red # Red if namespace is "kube-system"
  ['\.k8s\.local \(system\)$']=red # Red if context name ends with ".k8s.local" _and_ namespace is "system"
  ['^test-[0-9]+$']=yellow # Yellow if the entire content is "test-" followed by digits, and no namespace is displayed
)
[18:34:11] spaceship-prompt <k:dev11-system> $ echo ${(kv)SPACESHIP_KUBECONTEXT_COLORGROUPS}
^test-[0-9]+$ yellow \.k8s\.local \(system\)$ red \(kube-system\)$ red dev-01 green
```

This means that one has to be quite exact about the patterns.

Use case is that you might want to first define a namespace match, then a narrow context pattern and then a more general context name. But there is no guarantee that it'll work correctly :\

Reply:
Using value pairs with a normal array can solve this, although it's not as resilient...
```zsh
typeset -a SPACESHIP_KUBECONTEXT_COLORGROUPS=(
  dev-01                      green
  '\(kube-system\)$'          red
  '\.k8s\.local \(system\)$'  red
  '^test-[0-9]+$'             yellow
)
```

Reply:
Uh you are right, and I think having the priority order would be _very_ useful, so let's do a normal array with pairs.

Reply:
Aye sir! ;)

Reply:
I'll also reverse the order so color is first, this will make it much cleaner since color names are not of arbitrary length, as the patterns are.

Reply:
As you wish, I don't have a strong preference. The current approach reminds me more of a hashmap where keys are unique, but since it won't be a hashmap anyway, it doesn't really matter.

Reply:
My thought exactly. Pushing a new version now.